extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/TenantPolicyConfigResolver.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.keycloak.pep;
+
+import java.util.function.Supplier;
+
+import io.quarkus.keycloak.pep.runtime.KeycloakPolicyEnforcerTenantConfig;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A tenant resolver is responsible for resolving the {@link KeycloakPolicyEnforcerTenantConfig} for tenants, dynamically.
+ */
+public interface TenantPolicyConfigResolver {
+
+    /**
+     * Returns a {@link KeycloakPolicyEnforcerTenantConfig} given a {@code RoutingContext} and tenant id.
+     *
+     * @param routingContext the routing context
+     * @param tenantId tenant id
```

Comment:
I was think earlier, how this resolver aligns with the OIDC tenant resolution logic. Is this tenantId representing an OIDC tenant id resolved at the OIDC level ?

Reply:
Yep, I didn't change logic behind where this id comes from (just moved it around). This is current main:

https://github.com/quarkusio/quarkus/blob/243f46c10c4ac0cb380af120d942a9559a075b82/extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/runtime/KeycloakPolicyEnforcerAuthorizer.java#L63

Reply:
_I think_ (not sure) I only called it `tenantId` because in current main, it is kinda shared tenant, you select it `quarkus.keycloak."tenant".connection-pool-size` so it is OIDC tenant, but it is also Keycloak tenant. If there is such a thing. I suppose there is at least theoretically.

Reply:
I was thinking about it, IMHO it makes sense to pass `OidcTenantConfig` representing the current tenant since here it is a custom code, what will it do with this tenant id alone, not clear. FYI, DefaultTenantConfigResolver has a helper like  `getResolvedTenant(String tenantId)`

Reply:
I think your ideas are good. I know what they would do with the tenant id (identity tenant exactly as we do, via id so that we know which config belongs to him), but having whole config is much more comfortable for the user.